product_supplierinfo_for_customer_sale/views/sale_view.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+<data>
+
+    <record id="sale_order_form_context_supplierinfo" model="ir.ui.view">
+        <field name="name">sale.order.form.context.supplierinfo</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/form//field[@name='product_id']" position="attributes">
```

Comment:
Add a new xpath for `//field[@name='order_line']/tree//field[@name='product_id']` too (Notice the `tree` instead of `form` both xpath are required)